### PR TITLE
feat: BREAKING integer points

### PR DESCRIPTION
**Pull request by sentientwaffle**
* Use BigNumber for internal math. Use integers for points.
* Serialize route points as base64-encoded String instead of an Array of Number.
* Don't include the `liquidityCurve` on `RoutingTables#findBestHopFor{Source,Destination}Amount()`
* add `LiquidityCurve.{toBuffer,fromBuffer}`

### CODE REVIEW

**Comment by michielbdejong** [comment]
Retrying https://circleci.com/gh/interledgerjs/ilp-kit/1938 now

File: test/liquidity-curve.test.js [hunk 1]
```diff
@@ -8,45 +8,59 @@ describe('LiquidityCurve', function () {
     it('saves the points', function () {
       const points = [ [1, 2], [3, 4] ]
       const curve = new LiquidityCurve(points)
-      assert.deepEqual(curve.points, points)
-    })
-  })
-
-  describe('setPoints', function () {
```
Comment (michielbdejong) [inline test/liquidity-curve.test.js]: If `setPoints` is replaced by `setData`, then removing this test is correct, but shouldn't a new one for `setData`  be added in its place?

Comment (sentientwaffle) [inline test/liquidity-curve.test.js]: It's covered in the `LiquidityCurve` constructor tests.

File: test/liquidity-curve.test.js [hunk 2]
```diff
@@ -105,91 +119,6 @@ describe('LiquidityCurve', function () {
   })
 
   describe('combine', function () {
-    it('doesnt create non-increasing y coordinates', function () {
```
Comment (michielbdejong) [inline test/liquidity-curve.test.js]: Can't we convert these tests to the new data format and keep them, instead of removing them? It took @momerath42 and me a lot of work to find these edge case examples :)

File: test/route.test.js [hunk 3]
```diff
@@ -57,10 +57,6 @@ describe('Route', function () {
       it('finds the corresponding amount', function () {
         assert.equal(route.amountAt(55), 110)
       })
-
-      it('returns 0 if the actual value is negative', function () {
```
Comment (michielbdejong) [inline test/route.test.js]: are there a tests for the new `bnFromValue` function?

File: test/routing-tables.test.js [hunk 4]
```diff
@@ -630,10 +631,13 @@ describe('RoutingTables', function () {
           finalLedger: ledgerB,
           finalAmount: '50',
           minMessageWindow: 1,
-          additionalInfo: {rate_info: '0.5'},
-          liquidityCurve: [ [0, 0], [200, 100] ]
+          additionalInfo: {rate_info: '0.5'}
         })
     })
+
+    it('always returns an integer', function () {
```
Comment (michielbdejong) [inline test/routing-tables.test.js]: more descriptive maybe, "a string integer destinationAmount"

File: test/liquidity-curve.test.js (hunk 2 above)
Comment (sentientwaffle) [inline test/liquidity-curve.test.js]: Since the coordinates are integers now I'm not sure how that would work.

File: test/liquidity-curve.test.js (hunk 1 above)
Comment (michielbdejong) [inline test/liquidity-curve.test.js]: ah ok

File: test/liquidity-curve.test.js (hunk 2 above)
Comment (michielbdejong) [inline test/liquidity-curve.test.js]: hm, I guess... we can readd them if we find new live bugs. :)
